packages/components/src/confirm-dialog/types.ts

```diff
@@ -10,6 +10,8 @@ export type DialogInputEvent =
 type BaseProps = {
 	children: ReactNode;
 	onConfirm: ( event: DialogInputEvent ) => void;
+	confirmButtonText: string;
+	cancelButtonText: string;
```

Comment:
We should consider whether we want to allow only text, or if we want to be more permissive and allow any React component

Switching to allowing any React component (using the type `React.ReactNode`) would make this component more flexible, although in this case we may want to be more strict on purpose and only allow simple text in those buttons. What do folks think?

Reply:
For this type of basic dialog, basic text would be my vote.

Since we have fallback text, shouldn't we be marking these as optional? E.g.

```
confirmButtonText?: string;
cancelButtonText?: string;
```

Reply:
> For this type of basic dialog, basic text would be my vote.

That makes sense

> Since we have fallback text, shouldn't we be marking these as optional? E.g.

Absolutely, I must have missed it during my review. Thank you for flagging it!

Reply:
> Since we have fallback text, shouldn't we be marking these as optional?

Good call, thanks! Pushed 750e4d3